plate_mapper/plate_mapper.py

```diff
@@ -7,20 +7,48 @@
 # ----------------------------------------------------------------------------
 
 
+import sys
 import argparse
 
 
-def plate_mapper(input_f, barseq_f, output_f):
+def _print_list(l):
+    """ Print a list of strings in a human-friendly way
+
+    Parameter
+    ---------
+    l : list of str
+        Input list of strings
+
+    Return
+    ------
+    str
+        Output string
+
+    Notes
+    -----
+    If the list has <= 10 items, the entire list will be printed.
+    If the list has > 10 items, only the first three items will be printed,
+    followed by "... (no._of_items in total)"
+    """
+    if len(l) > 10:
+        return(', '.join(l[:3]) + '... (' + str(len(l)) + ' in total)')
+    else:
+        return(', '.join(l))
+
+
+def plate_mapper(input_f, barseq_f, output_f, names_f=None):
```

Comment:
This whole process is sequential and without reusing any part of the code. So I am not sure if modularity will be beneficial here. But it may be worth to separate "validate" as an independent function, so that it is easier to unit-test it. What do you think?